File: app/src/main/java/com/fasheep/boardgamehelper/Resource.java
package com.fasheep.boardgamehelper;

import java.util.HashMap;
import java.util.Map;

public class Resource {
    private static final Map<String, Integer> map = new HashMap<>();
    public static final String URL_UK = "http://fasheep.uksouth.cloudapp.azure.com";
    public static final String URL_CN = "http://120.78.126.136";

    static {
        map.put("wolf", R.drawable.wolf);
        map.put("hunter", R.drawable.hunter);
        map.put("seer", R.drawable.seer);
        map.put("witch", R.drawable.witch);
        map.put("guard", R.drawable.guard);
        map.put("idiot", R.drawable.idiot);
        map.put("villager", R.drawable.villager);
        map.put("wolf_king", R.drawable.wolf_king);

        map.put("defRoleImage", R.drawable.ic_role_image);

        map.put("avalon", R.drawable.avalon);
        map.put("custom", R.drawable.cunstom);
    }

    public static int getID(String id) {
        if (map.containsKey(id)) {
            return map.get(id);
        } else {
            return R.drawable.ic_role_image;
        }
    }
}

File: app/src/main/java/com/fasheep/boardgamehelper/res/Resource.kt
package com.fasheep.boardgamehelper.res

import com.fasheep.boardgamehelper.R

object Resource {
    private val map: MutableMap<String, Int> = HashMap()
    const val URL_HK: String = "http://bghhk.fasheep.xyz"

    init {
        map["wolf"] = R.drawable.wolf
        map["hunter"] = R.drawable.hunter
        map["seer"] = R.drawable.seer
        map["witch"] = R.drawable.witch
        map["guard"] = R.drawable.guard
        map["idiot"] = R.drawable.idiot
        map["villager"] = R.drawable.villager
        map["wolf_king"] = R.drawable.wolf_king

        map["defRoleImage"] = R.drawable.ic_role_image

        map["avalon"] = R.drawable.avalon
        map["custom"] = R.drawable.cunstom
    }

    fun getID(id: String?): Int {
        return if (id != null && map.containsKey(id)) {
            map[id]!!
        } else {
            R.drawable.ic_role_image
        }
    }
}
